src/app/clusters/webrtc-requestor-server/webrtc-requestor-server.cpp

```diff
@@ -0,0 +1,418 @@
+
+    if (mDelegate == nullptr)
+    {
+        ChipLogError(Zcl, "No delegate set for WebRTCRequestorServer");
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::Failure);
+        return;
+    }
+
+    switch (ctx.mRequestPath.mCommandId)
+    {
+    case Commands::Offer::Id:
+        CommandHandlerInterface::HandleCommand<Commands::Offer::DecodableType>(
+            ctx, [this](HandlerContext & ctx_, const auto & req) { HandleOffer(ctx_, req); });
+        break;
+    case Commands::Answer::Id:
+        CommandHandlerInterface::HandleCommand<Commands::Answer::DecodableType>(
+            ctx, [this](HandlerContext & ctx_, const auto & req) { HandleAnswer(ctx_, req); });
+        break;
+    case Commands::ICECandidates::Id:
+        CommandHandlerInterface::HandleCommand<Commands::ICECandidates::DecodableType>(
+            ctx, [this](HandlerContext & ctx_, const auto & req) { HandleICECandidates(ctx_, req); });
+        break;
+    case Commands::End::Id:
+        CommandHandlerInterface::HandleCommand<Commands::End::DecodableType>(
+            ctx, [this](HandlerContext & ctx_, const auto & req) { HandleEnd(ctx_, req); });
+        break;
+
+    default:
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::UnsupportedCommand);
+        break;
+        break;
+    }
+}
+
+void WebRTCRequestorServer::HandleOffer(HandlerContext & ctx, const Commands::Offer::DecodableType & req)
+{
+    // Extract command fields from the request
+    auto sdpOffer           = req.sdp;
+    auto iceServers         = req.ICEServers;
+    auto iceTransportPolicy = req.ICETransportPolicy;
+    uint16_t sessionId      = req.webRTCSessionID;
+
+    NodeId peerNodeId           = GetNodeIdFromCtx(ctx.mCommandHandler);
+    FabricIndex peerFabricIndex = ctx.mCommandHandler.GetAccessingFabricIndex();
+
+    // Check if the session ID is in the requestor list
+    WebRTCSessionStruct * existingSession = FindSession(sessionId);
+
+    if (!existingSession)
+    {
+        // If sessionId is not null and does not match a value in current sessions.
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
+        return;
+    }
+
+    // Also check that the existing session belongs to the same PeerNodeID / Fabric
+    // If it doesn’t match, respond with CONSTRAINT_ERROR
+    if (peerNodeId != existingSession->peerNodeID || peerFabricIndex != existingSession->peerFabricIndex)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
+        return;
+    }
+
+    // Delegate processing
+    WebRTCSessionStruct outSession;
+    CHIP_ERROR err =
+        mDelegate->HandleOffer(sessionId, sdpOffer, iceServers, iceTransportPolicy, peerNodeId, peerFabricIndex, outSession);
+
+    // Convert the CHIP_ERROR to an Interaction Model status code
+    Protocols::InteractionModel::Status status = StatusIB(err).mStatus;
```

Comment:
AddStatus needs ClusterStatusCode instead of Protocols::InteractionModel::Status. Use Protocols::InteractionModel::ClusterStatusCode directly convert the CHIP_ERROR to ClusterStatusCode